agent/taskresource/firelens/firelens_unix.go

```diff
@@ -94,33 +106,38 @@ type FirelensResource struct {
 }
 
 // NewFirelensResource returns a new FirelensResource.
-func NewFirelensResource(cluster, taskARN, taskDefinition, ec2InstanceID, dataDir, firelensConfigType, region, networkMode string,
-	firelensOptions map[string]string, containerToLogOptions map[string]map[string]string, credentialsManager credentials.Manager,
-	executionCredentialsID string) (*FirelensResource, error) {
-	firelensResource := &FirelensResource{
-		cluster:                cluster,
-		taskARN:                taskARN,
-		taskDefinition:         taskDefinition,
-		ec2InstanceID:          ec2InstanceID,
-		firelensConfigType:     firelensConfigType,
-		region:                 region,
-		networkMode:            networkMode,
-		containerToLogOptions:  containerToLogOptions,
-		ioutil:                 ioutilwrapper.NewIOUtil(),
-		s3ClientCreator:        factory.NewS3ClientCreator(),
-		executionCredentialsID: executionCredentialsID,
-		credentialsManager:     credentialsManager,
-	}
+func NewFirelensResource(cluster, taskARN, taskDefinition, ec2InstanceID, dataDir, region, networkMode string,
+	firelensConfig *apicontainer.FirelensConfig, containerToLogOptions map[string]map[string]string,
+	credentialsManager credentials.Manager, executionCredentialsID string) (*FirelensResource, error) {
 
 	fields := strings.Split(taskARN, "/")
 	taskID := fields[len(fields)-1]
-	firelensResource.resourceDir = filepath.Join(filepath.Join(dataDir, "firelens"), taskID)
 
-	err := firelensResource.parseOptions(firelensOptions)
-	if err != nil {
-		return nil, errors.Wrap(err, "error parsing firelens options")
-	}
+	firelensResource := &FirelensResource{}
+	if firelensConfig.Version != "v2" {
+		firelensResource = &FirelensResource{
```

Comment:
since `firelensResource` is already initialized before the if condition, this is not needed. 
we can just set `firelensResource.cluster = cluster` and so on.